lib/Sema/CSFix.h

```diff
@@ -862,6 +866,25 @@ class AllowInvalidInitRef final : public ConstraintFix {
                                      ConstraintLocator *locator);
 };
 
+class AllowTupleTypeMismatch final : public ConstraintFix {
+  Type LHS, RHS;
+
+  AllowTupleTypeMismatch(ConstraintSystem &cs, Type lhs, Type rhs,
+                         ConstraintLocator *locator)
+      : ConstraintFix(cs, FixKind::AllowTupleTypeMismatch, locator), LHS(lhs),
+        RHS(rhs) {}
+
+public:
+  static AllowTupleTypeMismatch *create(ConstraintSystem &cs, Type lhs,
+                                        Type rhs, ConstraintLocator *locator);
+
+  std::string getName() const override {
+    return "allow invalid tuple destructuring";
```

Comment:
Maybe it makes more sense to talk about mismatch here rather than destructuring which is just a special case?